include/openPMD/backend/Attribute.hpp

```diff
@@ -192,8 +209,12 @@ getCast( Attribute const a )
         return DoConvert<bool, U>{}(pvalue_b);
     else
         throw std::runtime_error("getCast: unknown Datatype.");
+}
 
-    return U{};
+template< typename U >
+U Attribute::get() const
+{
+    return getCast< U >( Variant::getResource() );
```

Comment:
I would hate to cast silently here and not notify the user about possible loss of precision.
There already is a function that does this ([`warnWrongDtype()`](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/src/backend/Attributable.cpp#L282-L290)). I'll push a patch shortly that will use that function if the resource dtype and `U` differ.

Reply:
That's actually not what I would like as a user.
If there is an attribute "attr" I can RT check it's dtype. If I want to `get<>` it as something, I want to get it as exactly such.

Reply:
Let's have a quick chat on the philosophy of that method if you like :) Gitter? VC?
In case you are busy today, we can also chat on Monday. I have to leave here around noon.

Reply:
Offline resolution:
C++ requires a certain effort from the user to put the *type* inside the brackets. Thus, he should be well aware of what he is requesting.
Information about the datatype can be obtained at run-time. If non-casting behavior is desired, we shift this obligation to the user.

Reply:
> If non-casting behavior is desired, we shift this obligation to the user.

e.g. by checking the `dtype` at runtime, taking the `resource` directly or casting back to the underlying `auxiliary::Variant` that will throw.